src/App.tsx

```diff
@@ -4,57 +4,131 @@
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Errors } from './types/Errors';
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
 
-              <Loader />
+export const App = () => {
+  const [usersList, setUsersList] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [currentError, setCurrentError] = useState<Errors | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[] | null>(null);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
+  const [newCommentCreating, setNewCommentCreating] = useState<boolean>(false);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  useEffect(() => {
+    getUsers()
+      .then(setUsersList)
+      .catch(() => setCurrentError(Errors.UsersLoading));
+  }, []);
+
+  useEffect(() => {
+    if (!selectedUser) {
+      return;
+    }
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    setCurrentError(null);
+    setUserPosts(null);
+    setIsPostsLoading(true);
+
+    getPosts(selectedUser.id)
+      .then(response => {
+        if (Array.isArray(response)) {
+          setUserPosts(response);
+          if (response.length === 0) {
+            setCurrentError(Errors.NoPosts);
+          }
+        } else {
+          setCurrentError(Errors.PostsLoading);
+        }
+      })
+      .catch(() => setCurrentError(Errors.PostsLoading))
+      .finally(() => {
+        setIsPostsLoading(false);
+      });
+  }, [selectedUser]);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={usersList}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {isPostsLoading && !currentError && <Loader />}
+
+                {currentError === Errors.PostsLoading && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
```

Comment:
The error message for post loading is displayed when `currentError` is `Errors.PostsLoading`. Ensure that this condition is mutually exclusive with the loader to avoid confusing the user.